src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -187,5 +187,15 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
 
             return candidate;
         }
+
+        /// <summary>
+        /// Returns true if the base type is a record (has a valid Clone method).
+        /// </summary>
+        internal static bool BaseTypeIsRecordNoUseSiteDiagnostics(NamedTypeSymbol baseType)
```

Comment:
>BaseTypeIsRecordNoUseSiteDiagnostics [](http://example.com/codeflow?start=29&length=36)

1. It doesn't feel like this helper belongs in this class/file.
2. It doesn't look like we need this helper at all. Tit looks like existing `IsRecord` property will be sufficient, especially as we aren't interested in any use-site information from this check.